### ISSUE TITLE
Ticklabels of horizontal colorbar not aligned when using text.usetex

### ISSUE THREAD

**Issue by vigeesh** [matplotlib/matplotlib#11243]
### Bug report

**Bug summary**
The positive and negative values of the major ticklabels in a horizontal colorbar are slightly shifted from the baseline when using text.usetex.

**Code for reproduction**

```python
import numpy as np
import matplotlib.pyplot as plt
import matplotlib as mpl

#### Un/Comment this part 
mpl.rcParams.update({'text.usetex': True})
####

fig = plt.figure()
p = plt.imshow(np.random.random([20,20])-.5)
cb = plt.colorbar(p, orientation="horizontal")
plt.show()
```

**Actual outcome**

![with_usetex](https://user-images.githubusercontent.com/11772224/39998312-fe299df0-5785-11e8-9f8b-fcc7de205d24.png)

**Expected outcome**

The positive and negative values are on the same baseline in the default case (i.e. without text.usetex). 
![without_usetex](https://user-images.githubusercontent.com/11772224/39998331-0a7dd6e8-5786-11e8-9310-277e73a8bb8d.png)

**Matplotlib version**

  * Operating system: macOS High Sierra / CentOS 6.9
  * Matplotlib version: 2.2.2 / 2.0.2
  * Matplotlib backend: MacOSX / Qt5Agg
  * Python version: 3.6.5 / 3.6.1

**Comment by avinashmnit30** [comment]
You can use **\text{-}** from latex->amsmath package instead of "**-**". The problem seems to be with the conventional latex rendering of "**-**" sign.
Try this code:-
```
import numpy as np
import matplotlib.pyplot as plt
import matplotlib as mpl
plt.switch_backend('Qt5Agg')

mpl.rcParams.update({'text.usetex': True})
params = {'text.latex.preamble' : [r'\usepackage{amsmath}']}
plt.rcParams.update(params)
fig = plt.figure()
p=plt.imshow(np.random.random([20,20])-.5)
a=plt.colorbar(p, orientation="horizontal")


labels = []
# Replace '-' with 'amsmath>> \text{-}'
for num in a.ax.get_xticklabels():
    num = num._text
    labels.append(r'$\text{-}' + num[2:]  if num[1] == '-' else r'' + num) 
    
#print(labels)
a.ax.set_xticklabels(labels)
plt.savefig('new.png')

plt.show()
```

**Comment by ksunden** [comment]
Appears to have been fixed by #16366 in mpl v3.3.0
